### ISSUE TITLE
List of list ver 19 VS ver 20

### ISSUE THREAD

**Issue by guillemmr** [psf/black#1941]
After downloading the last black version, notice some style differences in arrays of arrays:

In ver 19.10b0:
```python
my_list = [
        [12, 30, 1,],
        [13, 35, 2,],
        [14, 40, 3,],
        [15, 45, 4,],
        [16, 50, 5,],
        [17, 55, 6,],
        [18, 0, 7,],
    ]
```
In ver 20.8b1 (blacking the previous list)
```python
    my_list = [
        [
            12,
            30,
            1,
        ],
        [
            13,
            35,
            2,
        ],
        [
            14,
            40,
            3,
        ],
        [
            15,
            45,
            4,
        ],
        [
            16,
            50,
            5,
        ],
        [
            17,
            55,
            6,
        ],
        [
            18,
            0,
            7,
        ],
    ]
```
Is this expected? 
Black docs are explicit about this types:
![image](https://user-images.githubusercontent.com/34071081/105632266-82387780-5e52-11eb-8c93-86fe4da85bdf.png)

Thanks in advance!

**Comment by xrisk** [comment]
It's because of your trailing commas. They signal to Black that more elements may be added to the list in the future, and thus it tries to put each element onto a separate line.

You can pass `--skip-magic-trailing-comma`, or remove the trailing commas.

In your case it would be like:

```
my_list = [
        [12, 30, 1],
        [13, 35, 2],
        [14, 40, 3],
        [15, 45, 4],
        [16, 50, 5],
        [17, 55, 6],
        [18, 0, 7],
    ]
```

**Comment by ichard26** [comment]
Black used to remove the trailing comma if the expression fits in a single line, but this was changed by #826 and #1288. Now a trailing comma tells Black to always explode the expression. This change was made mostly for the cases where you _know_ a collection or whatever will grow in the future. Having it always exploded as one element per line reduces diff noise when adding elements. Before the "magic trailing comma" feature, you couldn't anticipate a collection's growth reliably since collections that fitted in one line were ruthlessly collapsed regardless of your intentions. One of Black's goals is reducing diff noise, so this was a good pragmatic change.

So no, this is not a bug, but an intended feature. The reason why you're filing this issue is probably since we say the following in the (outdated) style documentation:
> Unnecessary trailing commas are removed if an expression fits in one line. This makes it 1% more likely that your line won't exceed the allotted line length limit. Moreover, in this scenario, if you added another argument to your call, you'd probably fit it in the same line anyway. That doesn't make diffs any larger.

We missed that this paragraph became incorrect when the "magic trailing comma" feature was introduced. It was eventually fixed in commit 6b935a34d08c0b18d858e7da5093dc69a29c55f2, but that was after the stable documentation was released alongside `Black 20.8b1`. 

Anyway, [here's the documentation](https://github.com/psf/black/blob/master/docs/the_black_code_style.md#the-magic-trailing-comma) on the "magic trailing comma". Hopefully that helps and sorry for the possible confusion.

**Comment by guillemmr** [comment]
All clear,
Thank you @xrisk and @ichard26!
